packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -34,37 +35,36 @@ import { unlock } from '../../lock-unlock';
 
 function BlockStylesPanel( { clientId } ) {
 	return (
-		<div>
-			<PanelBody title={ __( 'Styles' ) }>
-				<BlockStyles clientId={ clientId } />
-			</PanelBody>
-		</div>
+		<PanelBody title={ __( 'Styles' ) }>
+			<BlockStyles clientId={ clientId } />
+		</PanelBody>
 	);
 }
 
 function BlockInspectorLockedBlocks( { topLevelLockedBlock } ) {
+	const { getBlockName, getBlockEditingMode } = useSelect( blockEditorStore );
 	const { contentClientIds, hasBlockStyles } = useSelect(
 		( select ) => {
-			const {
-				getClientIdsOfDescendants,
-				getBlockName,
-				getBlockEditingMode,
-			} = select( blockEditorStore );
+			const { getClientIdsOfDescendants } = select( blockEditorStore );
 			const { getBlockStyles } = select( blocksStore );
 			return {
-				contentClientIds: getClientIdsOfDescendants(
-					topLevelLockedBlock
-				).filter(
-					( clientId ) =>
-						getBlockName( clientId ) !== 'core/list-item' &&
-						getBlockEditingMode( clientId ) === 'contentOnly'
-				),
+				contentClientIds:
+					getClientIdsOfDescendants( topLevelLockedBlock ),
 				hasBlockStyles: !! getBlockStyles(
 					getBlockName( topLevelLockedBlock )
 				)?.length,
 			};
 		},
-		[ topLevelLockedBlock ]
+		[ topLevelLockedBlock, getBlockName ]
+	);
+	const eligibleContentClientIds = useMemo(
+		() =>
+			contentClientIds.filter(
+				( clientId ) =>
+					getBlockName( clientId ) !== 'core/list-item' &&
+					getBlockEditingMode( clientId ) === 'contentOnly'
+			),
+		[ contentClientIds, getBlockName, getBlockEditingMode ]
```

Comment:
Using two `useSelect` subs might be better, one for getting `contentClientIds` (existing) and one for the block name.

The `useMemo` will only run the filter when `contentClientIds` changes. What happens when client IDs are the same but `getBlockEditingMode` changes?

Reply:
You're right, yes. You suggest to keep the old one for `contentClientIds` (hack) and one `useSelect` for `hasBlockStyles` or something different?

Reply:
> You suggest to keep the old one for contentClientIds (hack) and one useSelect for hasBlockStyles

Exactly! No idea where block name selection popped into my brain 😄